design-proposals/project-deploymnet.md

```diff
@@ -0,0 +1,169 @@
+
+* It consumes a simpler and more concise YAML configuration.
+* The use of CRD and the operator aids in automating deployment changes over time.
+* The proposal differentiates between fabric and site administrators and provides a more secure deployment model via k8s RBAC.
+
+### Impact and desired outcome
+
+Implementing this proposal is expected to improve and simplify the deployment process for the ClusterLink users.
+Aligning the deployment process with K8s standards will particularly benefit use cases involving central control management.
+
+### Prior discussion and links
+
+Not applicable.
+
+## User/User Story
+
+As a site administrator, I would like to join my cluster to an existing fabric. The fabric administrator should send me the certificates required to join the fabric: the fabric CA certificate and the cluster certificates (private and public keys). Using these certificates and the ClusterLink operator, I can deploy the ClusterLink system in my cluster. Now, any user in my cluster can connect to other remote services in the fabric, even if they are located on different locations, or different cloud providers.
+
+## Goals
+
+This design document should:
+
+* Define the steps for deploying a ClusterLink gateway to a K8s cluster.
+* Define the task functions of the ClusterLink deployment operator for setting up the ClusterLink components in the cluster.
+* Define the CRD (Custom Resource Definition) for the ClusterLink deployment operator.
+
+## Non-Goals
+
+The following aspects are explicitly excluded and are out of scope in the current design:
+
+* This document focuses only on the deployment of ClusterLink to the K8s cluster.
+The deployment to another environment as VMs is out of scope.
+* Security considerations related to the creation of certificates for the fabric or each site by the fabric administrator are also beyond the scope of this document.
+
+## Proposal
+
+The proposal describes how the ClusterLink project can be deployed on any K8s Cluster. In this proposal, we distinguish between two entities: first, the fabric administrator, who manages the fabric and is responsible for determining which clusters can join the fabric, and second, the site administrator, who manages the k8s Cluster and is responsible for deploying the ClusterLink into the cluster.
+
+Before deploying the ClusterLink, a few prerequisite steps should be completed:
+
+1. The fabric administrator should create the site certificates (public and private) and transfer them, along with the public CA certificate, to the site administrator.
+1. The site administrator should deploy the certificates as k8s secrets to the cluster.
+1. The site administrator should deploy the ClusterLink operator to the cluster.
+
+After completing all the prerequisites, the site administrator can deploy the ClusterLink Custom Resource Definition (CRD) to the operator. The operator will then create the ClusterLink components, including:
```

Comment:
There are two separate objects in play:
- the CRD (i.e., the class)
- a CRD object (i.e., the instance).

The class can/should be registered with the API server before the operator is deployed (since the operator will attempt to create a watch on the object, the object schema - or CRD - must already be defined). Both should be done before the site administrator can create the CRD instance that defines the desired deployment configuration)